src/bundle/Resources/public/js/scripts/core/tag.view.select.js

```diff
@@ -0,0 +1,144 @@
+(function(global, doc, eZ) {
+    class TagViewSelect {
+        constructor(config) {
+            this.inputSelector = config.inputSelector || 'input';
+            this.fieldContainer = config.fieldContainer;
+
+            if (!this.fieldContainer) {
+                throw new Error('Field Container doesn\'t exist!');
+            }
+
+            this.container = this.fieldContainer.querySelector('.ibexa-tag-view-select');
+            this.listContainer = this.container.querySelector('.ibexa-tag-view-select__selected_list');
+            this.inputField = this.fieldContainer.querySelector(this.inputSelector);
+            this.selectButton = this.container.querySelector('.ibexa-tag-view-select__btn-select-path')
+            this.singleSelect = this.container.dataset.singleSelect === '1';
+            this.canDeleteLast = this.container.dataset.canDeleteLast === '1';
+            this.inputSeparator = config.seperator || ',';
+            this.selectedItemTemplate = this.listContainer.dataset.template;
+
+            this.addItems = this.addItems.bind(this);
+            this.addItem = this.addItem.bind(this);
+            this.removeItems = this.removeItems.bind(this);
+            this.removeItem = this.removeItem.bind(this);
+            this.toggleDeleteButtons = this.toggleDeleteButtons.bind(this);
+            this.attachDeleteEvents = this.attachDeleteEvents.bind(this);
+            this.adjustButtonLabel = this.adjustButtonLabel.bind(this);
+            this.toggleDisabledState = this.toggleDisabledState.bind(this);
+
+            this.attachDeleteEvents();
+
+            this.disabledObserver = new MutationObserver((mutationsList) => {
+                const isDisabled = mutationsList[0].target.hasAttribute('disabled');
+
+                this.toggleDisabledState(isDisabled);
+            });
+
+            this.disabledObserver.observe(this.container, {
+                attributeFilter: ['disabled'],
+                attributeOldValue: true,
+            });
+        }
+
+        toggleDisabledState(disabledState) {
+            const removeBtns = this.listContainer.querySelectorAll('.ibexa-tag-view-select__selected_item_remove');
+
+            removeBtns.forEach((btn) => btn.toggleAttribute('disabled', disabledState));
```

Comment:
Have you considered styling disabled buttons with CSS and preventing clicking with `pointer-events: none` instead of using MutationObserver?